src/Chapter1.hs

```diff
@@ -641,7 +662,12 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+
+firstDigit :: Integral a => a -> a
+firstDigit n
+    | (abs n) < 10 = (abs n)
+    | (abs n) >= 10 = firstDigit (div (abs n) 10)
+    | otherwise = 0
```

Comment:
I guess the last case is redundant. You can write:

```suggestion
    | abs n < 10 = abs n
    | otherwise = firstDigit (div (abs n) 10)
```

You can also see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼

Reply:
I added otherwise = 0 because it was complaining that the cases were non-exhaustive. I slapped that on there and the message went away. Another case where I didn't really try to optimize after I made it work. 👍 I also see I had some unnecessary parentheses.